src/Backends/InterfaceSettings.vala

```diff
@@ -57,5 +78,44 @@ public class SettingsDaemon.Backends.InterfaceSettings : GLib.Object {
         accounts_service.cursor_size = interface_settings.get_int (CURSOR_SIZE);
         accounts_service.locate_pointer = interface_settings.get_boolean (LOCATE_POINTER);
         accounts_service.text_scaling_factor = interface_settings.get_double (TEXT_SCALING_FACTOR);
+
+        accounts_service.picture_options = background_settings.get_enum (PICTURE_OPTIONS);
+        accounts_service.primary_color = background_settings.get_string (PRIMARY_COLOR);
+    }
+
+    private void sync_background_to_greeter () {
+        // File.new_for_uri creates file with broken get_basename method, so do this
+        var wallpaper_uri = background_settings.get_string (PICTURE_URI);
+        var wallpaper_path = File.new_for_uri (wallpaper_uri).get_path ();
+        var source = File.new_for_path (wallpaper_path);
+
+        var greeter_data_dir = Path.build_filename (Environment.get_variable ("XDG_GREETER_DATA_DIR"), "wallpaper");
+        if (greeter_data_dir == "") {
+            greeter_data_dir = Path.build_filename ("/var/lib/lightdm-data/", Environment.get_user_name (), "wallpaper");
+        }
+        var folder = File.new_for_path (greeter_data_dir);
+        var dest = File.new_for_path (Path.build_filename (greeter_data_dir, source.get_basename ()));
+
+        try {
+            if (folder.query_exists ()) {
+                var enumerator = folder.enumerate_children ("standard::*", FileQueryInfoFlags.NOFOLLOW_SYMLINKS);
+                FileInfo? info = null;
+                while ((info = enumerator.next_file ()) != null) {
+                    enumerator.get_child (info).@delete ();
+                }
+            } else {
+                folder.make_directory_with_parents ();
+            }
+
+            source.copy (dest, FileCopyFlags.OVERWRITE | FileCopyFlags.ALL_METADATA);
+        } catch (Error e) {
+            warning (e.message);
+            return;
+        }
+
+        // Ensure wallpaper is readable by greeter user (owner rw, others r)
+        FileUtils.chmod (dest.get_path (), 0604);
+
+        display_manager_accounts_service.background_file = dest.get_path ();
```

Comment:
this should be done inside the try/catch block. and BackgroundFile unset if a error happens.

Reply:
It produces compilation warning `warning: unreachable catch clause detected`.
And how to unset a property? `display_manager_accounts_service.background_file = "";`?

Reply:
> It produces compilation warning warning: unreachable catch clause detected.

sorry, i think you misunderstand, it's to move this to the try/catch block that we use because of the `source.copy()` above, since we depend in that not throwing a error to this be valid.

> And how to unset a property? display_manager_accounts_service.background_file = "";?

yes, it would be it.